src/libflake/flake/flakeref.cc

```diff
@@ -88,7 +88,7 @@ std::pair<FlakeRef, std::string> parsePathFlakeRefWithFragment(
     if (fragmentStart != std::string::npos) {
         fragment = percentDecode(url.substr(fragmentStart+1));
     }
-    if (pathEnd != std::string::npos && fragmentStart != std::string::npos) {
+    if (pathEnd != std::string::npos && fragmentStart != std::string::npos && url[pathEnd] == '?') {
```

Comment:
It was possible to pass the fragment as a "query" with the following flakeRef.

```
.#bla
```

I don't think this was expected behavior, since the above doesn't have a query.